Tensile/KernelWriterAssembly.py

```diff
@@ -7741,7 +7720,7 @@ def localReadDo(self, kernel, bufferIdx, iui, epsi, uIdx, tP):
                   InstructionTileOuput = (globalParameters["WavefrontWidth"] // kernel["MIWG0"] ) * (kernel["MatrixInstN"] // kernel["InstSplit"])
                 blockOffset = (InstructionTileOuput) * kernel["DepthU"] * tP["bpe"]
                 ldsPadOffset = (blockOffset//kernel["LdsBlockSizePerPad"])*kernel["LdsPad%s"%tc]*tP["bpe"]
-                offset = (blockOffset + ldsPadOffset)*rIdx + uIdx*kernel["MatrixInstK"]*tP["bpe"] + tP["localReadSwapByteOffset"]
+                offset = (blockOffset + ldsPadOffset)*rIdx + (uIdx)*kernel["MatrixInstK"]*tP["bpe"] + tP["localReadSwapByteOffset"]
```

Comment:
The unroll address increment here (`uIdx`) shouldn't be needed. There's this `tP["localReadOffset"]` gone missing here doing exactly just that, which increases the offset for each loop

This is a valid refactoring when all instances of `tP["localReadOffset"]` are removed. But maintaining 2 ways of calculating things adds confusion IMHO. You can see that's why I removed `uIdx` from `localReadDo` in PR #871 
https://github.com/ROCmSoftwarePlatform/Tensile/pull/871/files#diff-c4acdc43110b6b9fd5fee5e02641c9ebR7562

Reply:
you are right.. having K into localReadDo() gives flexibility to read 'any k value'